test/jasmine/tests/cartesian_test.js

```diff
@@ -150,6 +150,52 @@ describe('restyle', function() {
                 expect(firstLine2).toBe(secondLine2);
             }).then(done);
         });
+
+        it('can change scatter mode', function(done) {
+            var mock = Lib.extendDeep({}, require('@mocks/text_chart_basic.json'));
+
+            function assertScatterModeSizes(lineSize, pointSize, textSize) {
+                var gd3 = d3.select(gd),
+                    lines = gd3.selectAll('g.scatter.trace .js-line'),
```

Comment:
Nice. Yeah. I was debating adding a bunch of d3 tests since technically it's testing internal behavior that's subject to change without affecting the look of the plots. But realistically, I think this is best.